gateway/backend_blocks.go

```diff
@@ -126,6 +120,74 @@ func (bb *BlocksBackend) Get(ctx context.Context, path path.ImmutablePath, range
 		ra = &ranges[0]
 	}
 
+	// For range requests on UnixFS files, we can optimize by only fetching
+	// the root block first to determine if it's a file, then create a
+	// UnixfsFile that will lazily fetch only the blocks needed for the range
+	if ra != nil {
```

Comment:
ℹ️ this is a fix for bug that was already in boxo/gateway, adding timeout only surfaced the problem in https://github.com/ipfs/rainbow/pull/285 (where we run against all three backends)